### ISSUE TITLE
Empty page warning with http status 304

### ISSUE THREAD

**Issue by TheCasualObserver** [rss2email/rss2email#240]
Debian 12 tested with both:
- rss2email 3.13.1
- [bc6bf6c]

When I try to access repeatedly https://factorio.com/blog/rss I get the warning:
```
2023-06-22 12:07:23,789 [WARNING] empty page: factorio (https://factorio.com/blog/rss -> email)
```

I am not sure if this is the right solution, but I was able to mitigate this issue by:
``` diff
diff --git a/rss2email/feed.py b/rss2email/feed.py
index 790c499..6c9b822 100644
--- a/rss2email/feed.py
+++ b/rss2email/feed.py
@@ -425,7 +425,7 @@ class Feed (object):
             if 'html' in http_headers.get('content-type', 'rss'):
                 _LOG.warning('looks like HTML: {}'.format(self))
                 warned = True
-            if http_headers.get('content-length', '1') == '0':
+            if http_headers.get('content-length', '1') == '0' and status != 304:
                 _LOG.warning('empty page: {}'.format(self))
                 warned = True
```

**Comment by popcorncx** [comment]
Ubuntu 22.04.3 LTS with rss2email 3.13.1 and adc16fd2d014b016b8033e07d62a5e657a213f33 (3.14)

I noticed this when I added the Nebula RSS feeds, eg https://rss.nebula.app/video/channels/legaleagle.rss

The etag from the previous request is being stored and then passed with the next request, so a 304 response with an empty body is legitimate.

I also noticed in the code the place where it warns about an empty page, but I'm not sure if suppressing the warning would be the right solution...

A bit higher up in _check_for_errors at line 414 it does a comparison allowing for 304 as a valid status, but I don't see any specific handling of it. Should it check for the 304 and then exit, not doing any of the other checks because by definition there is no change since the previous request?

(I know enough python to follow what this is doing, but I don't know enough python or how the remainder of rss2email is structured to know what the right change would be)
